File: repos_estudiantes_ALVAREZ_tercera_entrega/AJATA_CANAVIRI_NICOL_MAYRA/Persistencia_de_Objetos/Ejercicio3/Java/Cliente.java
package Persistencia_de_Objetos.Ejercicio3.Java;

public class Cliente {
    private int id;
    private String nombre;
    private int telefono;

    public Cliente(int id, String nombre, int telefono) {
        this.id = id;
        this.nombre = nombre;
        this.telefono = telefono;
    }

    public int getId() {
        return id;
    }

    public String getNombre() {
        return nombre;
    }

    public int getTelefono() {
        return telefono;
    }

    @Override
    public String toString() {
    
        return String.format("%d,%s,%d", id, nombre, telefono);
    }

    
    public static Cliente fromString(String clienteStr) {
        try {
            String[] partes = clienteStr.trim().split(",");
            if (partes.length == 3) {
                int id = Integer.parseInt(partes[0]);
                String nombre = partes[1];
                int telefono = Integer.parseInt(partes[2]);
                return new Cliente(id, nombre, telefono);
            } else {
                System.err.println("Advertencia: Formato de linea incorrecto para cliente: " + clienteStr);
                return null;
            }
        } catch (NumberFormatException e) {
            System.err.println("Error al parsear datos numericos de cliente '" + clienteStr + "': " + e.getMessage());
            return null;
        } catch (ArrayIndexOutOfBoundsException e) {
            System.err.println("Error de indice al parsear linea de cliente '" + clienteStr + "': " + e.getMessage());
            return null;
        }
    }
}

File: repos_estudiantes_ALVAREZ_tercera_entrega/AJATA_CANAVIRI_NICOL_MAYRA/Persistencia_de_Objetos/Ejercicio3/Python/Cliente.py

class Cliente:
   
    def __init__(self, id_cliente: int, nombre: str, telefono: int):
        self.id = id_cliente
        self.nombre = nombre
        self.telefono = telefono

    def __str__(self):
        
        return f"{self.id},{self.nombre},{self.telefono}"

    def __repr__(self):
        
        return self.__str__()

    @staticmethod
    def from_string(cliente_str: str):
        
        try:
            partes = cliente_str.strip().split(',')
            if len(partes) == 3:
                id_cliente = int(partes[0])
                nombre = partes[1]
                telefono = int(partes[2])
                return Cliente(id_cliente, nombre, telefono)
            else:
                print(f"Advertencia: Formato de linea incorrecto para cliente: {cliente_str}")
                return None
        except (ValueError, IndexError) as e:
            print(f"Error al parsear linea de cliente '{cliente_str}': {e}")
            return None

    def get_id(self) -> int:
        return self.id

    def get_nombre(self) -> str:
        return self.nombre

    def get_telefono(self) -> int:
        return self.telefono